game/src/main/java/game/player/PlayerCommandProcessor.java

```diff
@@ -0,0 +1,42 @@
+package game.player;
+
+import engine.command.Command;
+import game.exception.CorruptPlayerCommandException;
+import engine.exception.WrongObjectIdException;
+import engine.geometry.Direction;
+import game.command.TankMoveCommand;
+import game.object.Tank;
+
+public class PlayerCommandProcessor {
+
+    private final PlayerManager playerManager;
+
+    public PlayerCommandProcessor(PlayerManager playerManager) {
+        this.playerManager = playerManager;
+    }
+
+    public Command getCommand(PlayerCommand playerCommand) throws CorruptPlayerCommandException {
+        try {
+            Player player = playerManager.getPlayer(playerCommand.getPlayerId());
+
+            return getCommand(player.getTank(), playerCommand.getCommandId());
+        } catch (WrongObjectIdException e) {
+            throw new CorruptPlayerCommandException();
+        }
+    }
+
+    private Command getCommand(Tank tank, int commandId) {
```

Comment:
This method creates command not just get. 

And lets move this method to Player. We also can create some interface which has this method returning GameCommand. For example:

```
UserPlayer implements Player
AIPlayer implements Player (artificial intelligence)
```

and we may put Player interface to engine

Reply:
I'm not sure what interface Player would have, UserPlayer should have method like `EngineCommand createEngineCommand(PlayerCommandType type)`; while AI should have method like `EngineCommand createEngineCommand( " argument here that represent enviroment around AI" )`

Reply:
EngineCommand createEngineCommand(PlayerCommand playerCommand)


may be ?

```
playerCommand.getPlayerType()
playerCommand.getCommandType()
```